src/NLog/Targets/ColoredConsoleTarget.cs

```diff
@@ -292,43 +317,8 @@ private void Output(LogEventInfo logEvent, string message)
 
             try
             {
-                ConsoleRowHighlightingRule matchingRule = null;
-
-                foreach (ConsoleRowHighlightingRule cr in this.RowHighlightingRules)
```

Comment:
This foreach-loop will generate a memory allocation because of the GeEnumerator call on IList<T>. Under normal circumstances it is fine, but since this is in the hot path to writing an log entry we should be careful about allocations. This issue has been addressed in other targets by using a for-loop instead. Take a look at #1020.